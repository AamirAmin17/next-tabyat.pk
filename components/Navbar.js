import React from "react";
import Image from "next/image";
import tabyatLogo from "../assets/images/tabiyat_white_logo2.svg";
import NavbarStyles from "./Navbar.module.scss";
import { getDecryptLocalStorage } from "../localStorageWrapper";

function Navbar() {
  const [scroll, setScroll] = React.useState(false);

  React.useEffect(() => {
    window.addEventListener("scroll", () => {
      if (window.scrollY > 5) setScroll(true);
      if (window.scrollY < 5) setScroll(false);
    });
  }, []);

  return (
    <div style={{ minHeight: "200vh" }} className="mainClass">
      <nav className="min-h-[64px] bg-primary fixed w-full">
        <div className="flex items-center min-h-[64px] pl-5 z-10">
          <Image src={tabyatLogo} height={70} width={380} />
          <div className="h-14 flex">
            <div className="bg-[#F0F0F0] rounded-md text-sm p-1 ml-7 min-w-[160px] rounded-r-none relative">
              Deliverying to <br />
              <span className="text-[16px] font-medium">Karachi, ...</span>
              <div className="absolute right-5 top-3 text-2xl">
                <b>&#8681;</b>
              </div>
            </div>
            <div className="h-full">
              <input
                type="search"
                className="h-full border-0 outline-none pl-4 rounded-r-lg min-w-[550px]"
                placeholder="Search for products or categories"
              />
            </div>
          </div>

          <div className="flex w-full justify-end mr-4 text-secondaryWhite">
            <button
              className={`bg-primaryButton p-3 rounded-md px-8 ${
                scroll ? "hidden" : "block"
              }`}
              type="button"
              onClick={() => getDecryptLocalStorage("hello")}
            >
              Get started now
            </button>
            <button
              className={`font-medium mr-12 ${scroll ? "block" : "hidden"}`}
              type="button"
            >
              Login | Signup
            </button>
          </div>
        </div>

        <div
          className={`bg-primary ${
            NavbarStyles.scrollDiv
          } transition-all absolute w-full ${
            scroll ? "transform -translate-y-[55px] -z-10" : "z-0"
          }`}
          data-testid="scrollDiv"
        >
          <div className="ml-[200px] flex gap-6 text-secondaryWhite mt-auto h-full">
            <button type="button">Medicines</button>
            <button type="button">Wellness</button>
            <div
              className={`flex w-full justify-end text-secondaryWhite mr-16 ${
                scroll ? "hidden" : "block"
              }`}
            >
              <button className="font-medium" type="button">
                Login | Signup
              </button>
            </div>
          </div>
        </div>
      </nav>
    </div>
  );
}

export default Navbar;
