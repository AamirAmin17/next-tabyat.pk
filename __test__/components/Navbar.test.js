import { screen, render } from "@testing-library/react";
import Navbar from "../../components/Navbar";

describe("Testing the navbar component", () => {
  beforeEach(() => {
    render(<Navbar />);
  });
  test("testing the total buttons in navbar component as for now it is 5 buttons", async () => {
    const buttonList = await screen.findAllByRole("button");
    expect(buttonList).toHaveLength(5);
  });

  it("renders homepage unchanged", () => {
    const container = render(<Navbar />);
    expect(container).toMatchSnapshot();
  });
});
