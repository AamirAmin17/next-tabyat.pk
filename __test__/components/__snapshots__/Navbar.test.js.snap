// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the navbar component renders homepage unchanged 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mainClass"
        style="min-height: 200vh;"
      >
        <nav
          class="min-h-[64px] bg-primary fixed w-full"
        >
          <div
            class="flex items-center min-h-[64px] pl-5 z-10"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27380%27%20height=%2770%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <div
              class="h-14 flex"
            >
              <div
                class="bg-[#F0F0F0] rounded-md text-sm p-1 ml-7 min-w-[160px] rounded-r-none relative"
              >
                Deliverying to 
                <br />
                <span
                  class="text-[16px] font-medium"
                >
                  Karachi, ...
                </span>
                <div
                  class="absolute right-5 top-3 text-2xl"
                >
                  <b>
                    ⇩
                  </b>
                </div>
              </div>
              <div
                class="h-full"
              >
                <input
                  class="h-full border-0 outline-none pl-4 rounded-r-lg min-w-[550px]"
                  placeholder="Search for products or categories"
                  type="search"
                />
              </div>
            </div>
            <div
              class="flex w-full justify-end mr-4 text-secondaryWhite"
            >
              <button
                class="bg-primaryButton p-3 rounded-md px-8 block"
                type="button"
              >
                Get started now
              </button>
              <button
                class="font-medium mr-12 hidden"
                type="button"
              >
                Login | Signup
              </button>
            </div>
          </div>
          <div
            class="bg-primary scrollDiv transition-all absolute w-full z-0"
            data-testid="scrollDiv"
          >
            <div
              class="ml-[200px] flex gap-6 text-secondaryWhite mt-auto h-full"
            >
              <button
                type="button"
              >
                Medicines
              </button>
              <button
                type="button"
              >
                Wellness
              </button>
              <div
                class="flex w-full justify-end text-secondaryWhite mr-16 block"
              >
                <button
                  class="font-medium"
                  type="button"
                >
                  Login | Signup
                </button>
              </div>
            </div>
          </div>
        </nav>
      </div>
    </div>
    <div>
      <div
        class="mainClass"
        style="min-height: 200vh;"
      >
        <nav
          class="min-h-[64px] bg-primary fixed w-full"
        >
          <div
            class="flex items-center min-h-[64px] pl-5 z-10"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27380%27%20height=%2770%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <div
              class="h-14 flex"
            >
              <div
                class="bg-[#F0F0F0] rounded-md text-sm p-1 ml-7 min-w-[160px] rounded-r-none relative"
              >
                Deliverying to 
                <br />
                <span
                  class="text-[16px] font-medium"
                >
                  Karachi, ...
                </span>
                <div
                  class="absolute right-5 top-3 text-2xl"
                >
                  <b>
                    ⇩
                  </b>
                </div>
              </div>
              <div
                class="h-full"
              >
                <input
                  class="h-full border-0 outline-none pl-4 rounded-r-lg min-w-[550px]"
                  placeholder="Search for products or categories"
                  type="search"
                />
              </div>
            </div>
            <div
              class="flex w-full justify-end mr-4 text-secondaryWhite"
            >
              <button
                class="bg-primaryButton p-3 rounded-md px-8 block"
                type="button"
              >
                Get started now
              </button>
              <button
                class="font-medium mr-12 hidden"
                type="button"
              >
                Login | Signup
              </button>
            </div>
          </div>
          <div
            class="bg-primary scrollDiv transition-all absolute w-full z-0"
            data-testid="scrollDiv"
          >
            <div
              class="ml-[200px] flex gap-6 text-secondaryWhite mt-auto h-full"
            >
              <button
                type="button"
              >
                Medicines
              </button>
              <button
                type="button"
              >
                Wellness
              </button>
              <div
                class="flex w-full justify-end text-secondaryWhite mr-16 block"
              >
                <button
                  class="font-medium"
                  type="button"
                >
                  Login | Signup
                </button>
              </div>
            </div>
          </div>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mainClass"
      style="min-height: 200vh;"
    >
      <nav
        class="min-h-[64px] bg-primary fixed w-full"
      >
        <div
          class="flex items-center min-h-[64px] pl-5 z-10"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27380%27%20height=%2770%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
          <div
            class="h-14 flex"
          >
            <div
              class="bg-[#F0F0F0] rounded-md text-sm p-1 ml-7 min-w-[160px] rounded-r-none relative"
            >
              Deliverying to 
              <br />
              <span
                class="text-[16px] font-medium"
              >
                Karachi, ...
              </span>
              <div
                class="absolute right-5 top-3 text-2xl"
              >
                <b>
                  ⇩
                </b>
              </div>
            </div>
            <div
              class="h-full"
            >
              <input
                class="h-full border-0 outline-none pl-4 rounded-r-lg min-w-[550px]"
                placeholder="Search for products or categories"
                type="search"
              />
            </div>
          </div>
          <div
            class="flex w-full justify-end mr-4 text-secondaryWhite"
          >
            <button
              class="bg-primaryButton p-3 rounded-md px-8 block"
              type="button"
            >
              Get started now
            </button>
            <button
              class="font-medium mr-12 hidden"
              type="button"
            >
              Login | Signup
            </button>
          </div>
        </div>
        <div
          class="bg-primary scrollDiv transition-all absolute w-full z-0"
          data-testid="scrollDiv"
        >
          <div
            class="ml-[200px] flex gap-6 text-secondaryWhite mt-auto h-full"
          >
            <button
              type="button"
            >
              Medicines
            </button>
            <button
              type="button"
            >
              Wellness
            </button>
            <div
              class="flex w-full justify-end text-secondaryWhite mr-16 block"
            >
              <button
                class="font-medium"
                type="button"
              >
                Login | Signup
              </button>
            </div>
          </div>
        </div>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
